services/github/github-package-registry-version.service.js

```diff
@@ -0,0 +1,185 @@
+'use strict'
+
+const gql = require('graphql-tag')
+const Joi = require('@hapi/joi')
+const { renderVersionBadge } = require('../version')
+const { GithubAuthV4Service } = require('./github-auth-service')
+const { documentation, transformErrors } = require('./github-helpers')
+const { NotFound } = require('..')
+
+// https://developer.github.com/v4/object/registrypackageversion/
+const packageVersion = Joi.object({
+  preRelease: Joi.bool().required(),
+  version: Joi.string().required(),
+})
+
+const latestVersionSchema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      registryPackages: Joi.object({
+        nodes: Joi.array()
+          .items(
+            Joi.object({
+              latestVersion: packageVersion,
+            })
+          )
+          .min(0)
+          .required(),
+      }).required(),
+    }).required(),
+  }).required(),
+}).required()
+
+const versionSchema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      registryPackages: Joi.object({
+        nodes: Joi.array()
+          .items(
+            Joi.object({
+              versions: Joi.object({
+                nodes: Joi.array()
+                  .items(packageVersion)
+                  .min(0)
+                  .required(),
+              }),
+            })
+          )
+          .min(0)
+          .required(),
+      }).required(),
+    }).required(),
+  }).required(),
+}).required()
+
+module.exports = class GithubPackageRegistryVersion extends GithubAuthV4Service {
+  static get category() {
+    return 'version'
+  }
+
+  static get route() {
+    return {
+      base: 'github/packages',
+      pattern: ':type(v|vpre)/:user/:repo/:packageName',
```

Comment:
Either GitHub's Package Regisry does not yet correctly identify "prerelease" package versions, or the GraphQL is just always treating all package versions as non-pre 🤷‍♂ 

The end result, at least for now, is that we'll end up displaying the same version value (the one GitHub tells us is the latest) regardless of whether the user selects `v` or `vpre` for `type`